File: solutions/230.kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
const kthSmallest = (root,k)=>{
  let res = []

  const dfs= node => {
    if(!node) return

    res.push(node.val)

    dfs(node.left)
    dfs(node.right)
  }

  dfs(root)
  return res.sort((a,b)=>a-b)[k-1]
}

File: solutions/230.kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        in order traversal
        """
        stack = []
        li = []

        while stack or root:
            if root:
                stack.append(root)
                root = root.left
            else:
                cur = stack.pop()
                li.append(cur.val)
                root = cur.right

        return li[k - 1]


#         heap = []

#         def traverse(node):
#             if not node:
#                 return

#             heapq.heappush(heap, node.val)

#             traverse(node.left)
#             traverse(node.right)

#         traverse(root)

#         while k - 1:
#             heapq.heappop(heap)
#             k -= 1

#         return heapq.heappop(heap)
